examples/models/burgers_one2n.py

```diff
@@ -0,0 +1,276 @@
+        for Burgers equation. It is kept constant and should
+        be ignored by the network.
+        """
+        P0_2d = FunctionSpace(mesh, "DG", 0)
+        return Function(P0_2d).assign(1.0)
+
+    def drag(self, mesh):
+        """
+        Compute the bathymetry field on the current `mesh`.
+
+        Note that there isn't really a concept of bathymetry
+        for Burgers equation. It is kept constant and should
+        be ignored by the network.
+        """
+        P0_2d = FunctionSpace(mesh, "DG", 0)
+        return Function(P0_2d).assign(1.0)
+
+    def viscosity(self, mesh):
+        """
+        Compute the viscosity coefficient on the current `mesh`.
+        """
+        P0_2d = FunctionSpace(mesh, "DG", 0)
+        return Function(P0_2d).assign(self.viscosity_coefficient)
+
+    def ic(self, mesh):
+        """
+        Initial condition
+        """
+        x, y = SpatialCoordinate(mesh)
+        expr = self.initial_speed * sin(pi * x)
+        yside = self.initial_speed * sin(pi * y)
+        yside = 0
+        return as_vector([expr, yside])
+
+
+PETSc.Sys.popErrorHandler()
+parameters = Parameters()
+
+
+def get_function_space(mesh):
+    r"""
+    Construct the :math:`\mathbb P2` finite element space
+    used for the prognostic solution.
+    """
+    return VectorFunctionSpace(mesh, "CG", 2)
+
+
+class Solver(nn_adapt.solving.Solver):
+    """
+    Solver object based on current mesh and state.
+    """
+
+    def __init__(self, mesh, ic, **kwargs):
+        """
+        :arg mesh: the mesh to define the solver on
+        :arg ic: the current state / initial condition
+        """
+        self.mesh = mesh
+
+        # Collect parameters
+        dt = Constant(parameters.timestep)
+        nu = parameters.viscosity(mesh)
+
+        # Define variational formulation
+        V = self.function_space
+        u = Function(V)
+        u_ = Function(V)
+        v = TestFunction(V)
+        self._form = (
+            inner((u - u_) / dt, v) * dx
+            + inner(dot(u, nabla_grad(u)), v) * dx
+            + nu * inner(grad(u), grad(v)) * dx
+        )
+        problem = NonlinearVariationalProblem(self._form, u)
+
+        # Set initial condition
+        u_.project(parameters.ic(mesh))
+
+        # Create solver
+        self._solver = NonlinearVariationalSolver(problem)
+        self._solution = u
+
+    @property
+    def function_space(self):
+        r"""
+        The :math:`\mathbb P2` finite element space.
+        """
+        return get_function_space(self.mesh)
+
+    @property
+    def form(self):
+        """
+        The weak form of Burgers equation
+        """
+        return self._form
+
+    @property
+    def solution(self):
+        return self._solution
+
+    def iterate(self, **kwargs):
+        """
+        Take a single timestep of Burgers equation
+        """
+        self._solver.solve()
+        
+        
+class Solver_one2n(nn_adapt.solving.Solver):
```

Comment:
If this is largely the same as the `Solver` class in `examples.models.burgers` then you could just define this to be a subclass of it. That way, you only need to write out the methods that are different and don't need to repeat the others.